src/main/java/com/microsoft/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -115,6 +117,33 @@ public void failure(Throwable error) {
         });
     }
 
+    // Need to come back and implement callbacks instead of calling get(). 
+    // Could not find an example of the callback signature so postponing as this method
+    // will be primarily used as blocking call at app startup.
+    public Mono<List<SecretItem>> listSecrets(){
+        List<SecretItem> secrets=null;
+        try{
+            return Mono.just(_keyVaultClient.listSecretsAsync(getKeyVaultUri(), null).get());
+        }
+        catch(Exception exception)
+        {
+            return Mono.error(exception);
+        }
+    }
+
+    // For now, blocking to get MVP of call chain.  Switch to async in the future.
+    public Dictionary<String, String> getSecrets(){
+        List<SecretItem> secretItems = listSecrets().block();
+
+        Dictionary<String, String> secrets = new Hashtable<String, String>();
```

Comment:
Even though we are blocking, can we do a "waitAll" after firing off all of the getSecrets individually as "asyncs" instead of being Serial here? At least my interpretation is that this is serial. Application Startup time in production might not be super important but in dev it is uber painful as we launch the app again and again.

Reply:
There are more changes than just that.  The listsecrets call on the KV client needs to implement callbacks as well.  So, that's bigger piece of work.  As for local dev, it shouldn't be a problem, because the reality is that without access to the keys the app will fail.  So, if we adjust the calls, the startup of the app will still block() on the call, because if it doesn't you could end up with a race condition in execution.  Thus, I'm not even sure if we should change it or just make it a sync set of methods.

Reply:
Any reason to use Hashtable over HashMap? Hashtable is synchronous and is used for sharing data across threads.

Reply:
I didn't have a reason previously, but since you mention that, the object will be a Spring Singleton since it is marked with @Service.  So, it sounds like Hashtable would then be preferred?

Reply:
If there is a need for a thread-safe operation, then Hashtable can be used as all its methods are synchronized. As it's a legacy class better to avoid using it. If there's a need for multi-threaded, you can use ConcurrentHashMap which is almost similar to Hashtable.  Syncronized operations impacts performance so unless needed HashMaps are preferred.